src/vs/code/electron-sandbox/issue/issueReporterServiceNew.ts

```diff
@@ -0,0 +1,508 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { $, reset, windowOpenNoOpener } from 'vs/base/browser/dom';
+import { mainWindow } from 'vs/base/browser/window';
+import { Codicon } from 'vs/base/common/codicons';
+import { groupBy } from 'vs/base/common/collections';
+import { CancellationError } from 'vs/base/common/errors';
+import { isMacintosh } from 'vs/base/common/platform';
+import { ThemeIcon } from 'vs/base/common/themables';
+import { URI } from 'vs/base/common/uri';
+import { IssueReporterData as IssueReporterModelData } from 'vs/code/browser/issue/issueReporterModel';
+import { BaseIssueReporterService, hide, show } from 'vs/code/browser/issue/issue';
+import { localize } from 'vs/nls';
+import { isRemoteDiagnosticError } from 'vs/platform/diagnostics/common/diagnostics';
+import { IIssueMainService, IssueReporterData, IssueReporterExtensionData, IssueReporterWindowConfiguration, IssueType } from 'vs/platform/issue/common/issue';
+import { INativeHostService } from 'vs/platform/native/common/native';
+import { applyZoom, zoomIn, zoomOut } from 'vs/platform/window/electron-sandbox/window';
+
+// GitHub has let us know that we could up our limit here to 8k. We chose 7500 to play it safe.
+// ref https://github.com/microsoft/vscode/issues/159191
+const MAX_URL_LENGTH = 7500;
+
+
+export class IssueReporterNew extends BaseIssueReporterService {
```

Comment:
Is this a new version of an `IssueReporter`? If so, maybe calling it `IssueReporter2` instead would be better (like we do with `Action` and `Action2`) so that we can always make a new one in the future.